### PR TITLE
better fetching for initial state and all tabs

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

now that we're mving to local state we need to do better at fetching data

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-provider/src/hooks/use-transport.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useMemo } from "react";
```
Comment (rileylnapier) [inline packages/react-provider/src/hooks/use-transport.ts:1]: attempting to breakup some of the bigger components into smaller hooks

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 2]
```diff
@@ -180,14 +175,25 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
...
       event.preventDefault();
+      let myCurrentTab = currentTab;
       if (!isOpen) {
         setView("messages");
-        setCurrentTab(tabs?.[0] ?? DEFAULT_TABS[1]);
+        myCurrentTab = tabs?.[0] ?? DEFAULT_TABS[1];
+        setCurrentTab(myCurrentTab);
       }
 
+      if (!lastMessagesFetched) {
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Inbox/index.tsx:185]: will probably do something with this to "refetch" if a certain time has elapsed but can deal with that in later optimiation

File: packages/client-graphql/src/__tests__/messages.spec.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
...}\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n  list1: messages(params: $params1, limit: $limit) {\\\\n    totalCount\\\\n    pageInfo {\\\\n      startCursor\\\\n      hasNextPage\\\\n      __typename\\\\n    }\\\\n    nodes {\\\\n      id\\\\n      messageId\\\\n      created\\\\n      read\\\\n      tags\\\\n      content {\\\\n        title\\\\n        body\\\\n        blocks {\\\\n          ... on TextBlock {\\\\n            type\\\\n            text\\\\n            __typename\\\\n          }\\\\n          ... on ActionBlock {\\\\n            type\\\\n            text\\\\n            url\\\\n            __typename\\\\n          }\\\\n          __typename\\\\n        }\\\\n        data\\\\n        trackingIds {\\\\n          archiveTrackingId\\\\n          clickTrackingId\\\\n          deliverTrackingId\\\\n          readTrackingId\\\\n          unreadTrackingId\\\\n          __typename\\\\n        }\\\\n        __typename\\\\n      }\\\\n      __typename\\\\n    }\\\\n    __typename\\\\n  }\\\\n}\\\\n\\",\\"operationName\\":\\"GetMessages\\",\\"variables\\":{\\"params0\\":{\\"isRead\\":false},\\"params1\\":{\\"tags\\":[\\"abc\\",\\"123\\"]},\\"limit\\":10}}",
```
Comment (drew-y) [inline packages/client-graphql/src/__tests__/messages.spec.ts:157]: Beautiful

File: packages/client-graphql/src/brands.ts [hunk 4]
```diff
@@ -54,7 +54,7 @@ query GetBrand($brandId: String!) {
 `;
 
 const GET_INAPP_BRAND = `
-query GetInAppBrand {
+query GetInppBrand {
```
Comment (drew-y) [inline packages/client-graphql/src/brands.ts:57]: Was this change on purpose? I feel like this might be a typo.

File: packages/client-graphql/src/initial-state.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+interface IInitialStateParams {
+  skipFetchBrand?: boolean;
+  brandId?: string;
+  from?: number;
+}
+
+type GetInitialState = (params: IInitialStateParams) => Promise<
+  | {
+      brand: any;
```
Comment (drew-y) [inline packages/client-graphql/src/initial-state.ts:14]: Too bad we don't have types for this :/

File: packages/client-graphql/src/initial-state.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  | undefined
+>;
+
+const buildQuery = ({
+  skipFetchBrand,
+  brandId,
+}: Partial<IInitialStateParams>) => {
+  if (skipFetchBrand) {
+    return `
+      query InitialState($params: FilterParamsInput) {
```
Comment (drew-y) [inline packages/client-graphql/src/initial-state.ts:26]: Is it possible to use `directives` here in place of string interpolation here? Just thinking aloud.

Comment (drew-y) [reply packages/client-graphql/src/initial-state.ts:26]: Apparently not :/

Comment (drew-y) [reply packages/client-graphql/src/initial-state.ts:26]: Only supports fields.

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 7]
```diff
@@ -96,14 +96,17 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
   }, [props]);
 
   const tabs = props.tabs == false ? undefined : props.tabs;
```
Comment (drew-y) [inline packages/react-inbox/src/components/Inbox/index.tsx:98]:
```suggestion
  const tabs = props.tabs === false ? undefined : props.tabs;
```

File: packages/client-graphql/src/brands.ts (hunk 4 above)
Comment (drew-y) [inline packages/client-graphql/src/brands.ts:57]:
```suggestion
query GetInAppBrand {
```

**Review by drew-y — APPROVED** [review]
